onnxscript/rewriter/pattern.py

```diff
@@ -1371,7 +1371,11 @@ def try_rewrite(
                 if var.name is not None:
                     if var.name not in match.bindings:
                         match.bindings[var.name] = None
-            if not self._condition_function(context, **match.bindings):
+            check_match_result = self._condition_function(context, **match.bindings)
+            if not check_match_result:
+                # If check function was provided, but it failed, return the reason for failure to the tracer.
+                if not isinstance(check_match_result, bool):
```

Comment:
nit: changing the condition to `isinstance(check_match_result, MatchResult)` may be more robust